python/cuml/cluster/dbscan.pyx

```diff
@@ -49,20 +52,31 @@ cdef extern from "dbscan/dbscan_c.h" namespace "ML":
 
 class DBSCAN:
     """
-    Create a DataFrame, fill it with data, and compute DBSCAN:
+    DBSCAN is a very powerful yet fast clustering technique that finds clusters where
+    data is concentrated. This allows DBSCAN to generalize to all problems, since in
```

Comment:
I am very hesitant to make a statement in our documentation that DBSCAN generalizes to `all` problems. We don't get a free lunch in data science. DBSCAN uses density to overcome some of the shortcomings of k-means, without having to assume that the clusters can be modeled by Gaussian balls.

It does, however, still have the shortcoming that all clusters are assumed to have similar densities. Another shortcoming is that a good distance metric can be hard to find. There are other algorithms that overcome these particular shortcomings but those algorithms also introduce their own. It might benefit us to put these types of facts in our class-level documentation and try not to sound like we're overgeneralizing the usefulness.

Reply:
Oh ok. Sorry I was over-generalizing loll. Ok I will edit this to make it sound less one-sided.
I'll also mention the shortcomings.